vault/auth.go

```diff
@@ -148,6 +155,45 @@ func (v *Vault) UserPassLogin() string {
 	return secret.Auth.ClientToken
 }
 
+// EC2Login - AWS EC2 auth backend
+func (v *Vault) EC2Login() string {
+	role := env.Getenv("VAULT_AUTH_AWS_ROLE")
+	mount := env.Getenv("VAULT_AUTH_AWS_MOUNT", "aws")
+
+	vars := map[string]interface{}{}
+
+	if role != "" {
+		vars["role"] = role
+	}
+
+	opts := aws.ClientOptions{
+		Timeout: time.Duration(typeconv.MustAtoi(os.Getenv("AWS_TIMEOUT"))) * time.Millisecond,
+	}
+
+	meta := aws.NewEc2Meta(opts)
+
+	if endpoint := env.Getenv("AWS_META_ENDPOINT"); endpoint != "" {
+		meta.Endpoint = endpoint
```

Comment:
I think this block belongs here instead: https://github.com/hairyhenderson/gomplate/blob/master/aws/ec2meta.go#L89